### PR TITLE
update ReadAligner to modern Python class + docs

### PR DESCRIPTION
**Pull request by mr-c**
- [x] document CPython checklist

### CODE REVIEW

**Comment by mr-c** [comment]
@camillescott can you review the checklist?

File: khmer/_khmermodule.cc [hunk 1]
```diff
@@ -4396,6 +4402,9 @@ init_khmer(void)
 
     Py_INCREF(&khmer_KLabelHashType);
     PyModule_AddObject(m, "_LabelHash", (PyObject *)&khmer_KLabelHashType);
+
+    Py_INCREF(&khmer_ReadAlignerType);
+    PyModule_AddObject(m, "ReadAligner", (PyObject *)&khmer_ReadAlignerType);
```
Comment (camillescott) [inline khmer/_khmermodule.cc]: Standard practice thus far has been to define the internal CPython types as `_<name>`, then import them into Python-land as `<name>`.

**Comment by mr-c** [comment]
- [ ] the CPython object name should be of the form `khmer_${OBJECTNAME}_Object` (courtesy @luizirber )

File: doc/dev/coding-guidelines-and-review.txt [hunk 2]
```diff
@@ -75,3 +75,32 @@ ready for review::
...
+Here's a checklist for new CPython types::
+
+   - [ ] Named struct with PyObject_HEAD macro
+   - [ ] `static PyTypeObject khmer_${OBJECTNAME}_Type` with the following
+     entries
+      - [ ] `PyObject_HEAD_INIT(NULL)`
+      - [ ] all fields should have their name in a comment for readability
+      - [ ] The `tp_name` filed is a dotted name with both the module name and
+        the name of the type within the module. Example: `khmer.ReadAligner`
+      - [ ] Deallocator defined and cast to `(destructor)` in tp_dealloc
```
Comment (mr-c) [inline doc/dev/coding-guidelines-and-review.txt]: - [ ] The object's deallocator must be `self->ob_type->tp_free(self);`

**Comment by mr-c** [comment]
- [ ] Is it mergable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?

**Comment by mr-c** [comment]
Ready for review & merge @ctb @luizirber @camillescott

File: doc/dev/coding-guidelines-and-review.txt (hunk 2 above)
Comment (luizirber) [inline doc/dev/coding-guidelines-and-review.txt]: Maybe ```Py_TYPE(obj)->tp_free((PyObject*)obj);```

Reference: http://python3porting.com/cextensions.html#object-initialization

File: doc/dev/coding-guidelines-and-review.txt [hunk 3]
```diff
@@ -76,3 +76,34 @@ ready for review::
...
+CPython Checklist
+-----------------
+
+Here's a checklist for new CPython types::
+
+   - [ ] the CPython object name is of the form `khmer_${OBJECTNAME}_Object`
+   - [ ] Named struct with `PyObject_HEAD` macro
+   - [ ] `static PyTypeObject khmer_${OBJECTNAME}_Type` with the following
+     entries
+      - [ ] `PyObject_HEAD_INIT(NULL)`
```
Comment (luizirber) [inline doc/dev/coding-guidelines-and-review.txt]: We can use ```PyVarObject_HEAD_INIT(NULL, 0)``` and help on the Python 2/3 compat.

Reference: http://python3porting.com/cextensions.html#object-initialization

**Comment by mr-c** [comment]
@luizirber ready for merging; thanks!

**Pull request merged**